src/MongoDB.Driver.Core/Core/Authentication/External/IExternalAuthenticationCredentialsProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MongoDB.Driver.Core.Authentication.External
+{
+    internal interface IExternalAuthenticationCredentialsProvider<TCredentials> where TCredentials : IExternalCredentials
+    {
+        TCredentials CreateCredentialsFromExternal(CancellationToken cancellationToken = default);
+        Task<TCredentials> CreateCredentialsFromExternalAsync(CancellationToken cancellationToken = default);
+    }
```

Comment:
In the previous code before this refactoring, I believe we changed the terminology to `ExternalSource` for clarity rather than just `External`.